comid/integrityregisters_test.go

```diff
@@ -0,0 +1,298 @@
+	actual, err := reg.MarshalCBOR()
+	require.NoError(t, err)
+	fmt.Printf("CBOR Payload = %x\n", actual)
+	assert.Equal(t, expected, actual)
+}
+
+func TestIntegrityRegisters_UnmarshalCBOR_UIntIndex_OK(t *testing.T) {
+	expected := IntegrityRegisters{map[IRegisterIndex]Digests{
+		uint64(0): []swid.HashEntry{{HashAlgID: swid.Sha256, HashValue: MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")}},
+		uint64(1): []swid.HashEntry{{HashAlgID: swid.Sha256, HashValue: MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")}},
+		uint64(2): []swid.HashEntry{{HashAlgID: swid.Sha256, HashValue: MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")}},
+		uint64(3): []swid.HashEntry{{HashAlgID: swid.Sha256, HashValue: MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")}},
+		uint64(4): []swid.HashEntry{{HashAlgID: swid.Sha256, HashValue: MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")}},
+	}}
+	bstr := MustHexDecode(nil, `a5008182015820e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75018182015820e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75028182015820e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75038182015820e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75048182015820e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75`)
+	actual := IntegrityRegisters{}
+	err := actual.UnmarshalCBOR(bstr)
+	require.NoError(t, err)
+	require.True(t, reflect.DeepEqual(expected, actual))
+}
+
+func TestIntegrityRegisters_MarshalJSON_UIntIndex_OK(t *testing.T) {
+	expected := `{
+		"0": {
+			"key_type": "uint",
+			"Value": [
+				"sha-256;5Fty9cDAtXLbTY06t+l/No/3TmI0eoJN7LZ6hOUiTXU="
+			]
+		},
+		"1": {
+			"key_type": "uint",
+			"Value": [
+				"sha-256;5Fty9cDAtXLbTY06t+l/No/3TmI0eoJN7LZ6hOUiTXU="
+			]
+		},
+		"2": {
+			"key_type": "uint",
+			"Value": [
+				"sha-256;5Fty9cDAtXLbTY06t+l/No/3TmI0eoJN7LZ6hOUiTXU="
+			]
+		},
+		"3": {
+			"key_type": "uint",
+			"Value": [
+				"sha-256;5Fty9cDAtXLbTY06t+l/No/3TmI0eoJN7LZ6hOUiTXU="
+			]
+		},
+		"4": {
+			"key_type": "uint",
+			"Value": [
+				"sha-256;5Fty9cDAtXLbTY06t+l/No/3TmI0eoJN7LZ6hOUiTXU="
+			]
+		}
+	}
+	`
+	reg, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+	actual, err := reg.MarshalJSON()
+	require.NoError(t, err)
+	assert.JSONEq(t, expected, string(actual))
+}
+
+func TestIntegrityRegisters_MarshalCBOR_TextIndex_OK(t *testing.T) {
+	reg, err := prepareRegister(t, "text")
+	// A5                                   # map(5)
+	// 61                                   # text(1)
+	//    30                                # "0"
+	// 81                                   # array(1)
+	//    82                                # array(2)
+	// 	      01                             # unsigned(1)
+	//  	  58 20                          # bytes(32)
```

Comment:
we can remove this I think

Reply:
Resolved after adding clarification as to what this reflect and its significance!